assets/aml-benchmark/components/dataset-downloader/spec.yaml

```diff
@@ -9,19 +9,19 @@ version: 0.0.1
 inputs:
   dataset_name: 
     type: string
-    description: Name of the dataset to download from HuggingFace; must be null if url is specified.
+    description: Name of the dataset to download from HuggingFace; must be null if script is specified.
     optional: True
   configuration:
     type: string
-    description: If a specific sub-dataset of the HuggingFace dataset to download, specify the configuration name; specify 'all' to download all configurations. Else, leave it null.
+    description: If a specific sub-dataset of the dataset to download, specify the configuration name; specify 'all' to download all configurations. Else, leave it null.
     optional: True
   split:
     type: string
-    description: If a specific split of the HuggingFace dataset to download, specify the split name; specify 'all' to download all splits. Leave it null if url is specified.
-    optional: True
-  url:
-    type: string
-    description: URL of the dataset to download. Must be null if dataset_name is specified.
+    description: If a specific split of the dataset to download, specify the split name; specify 'all' to download all splits.
+    optional: False
+  script_path:
+    type: uri_file
+    description: Path to the dataset loading script. Must follow the HuggingFace dataset loading script template.
```

Comment:
Do we need to provide the link to the sample template? Also, are we allowed to add such links to external sources?

Reply:
That's exactly the reason I did not include the link. @EricWrightAtWork, do we have clarity on this, your thoughts?